### PR TITLE
Fix entropy function

### PR DESCRIPTION
**Pull request by ekaterinats**
Entropy function was incorrect, now it should be fixed.

### CODE REVIEW

**Comment by timcallow** [comment]
@DanielKotik I can confirm these changes are ok from the "physics perspective" but I'm not sure why one of the checks is failing.

**Comment by DanielKotik** [comment]
> @DanielKotik I can confirm these changes are ok from the "physics perspective" but I'm not sure why one of the checks is failing.

I'll have a look.

File: atoMEC/mathtools.py [hunk 1]
```diff
@@ -194,6 +239,46 @@ def fd_int_complete(mu, beta, n):
...
+    -------
+    I_n : float
+        the complete fermi-dirac integral
+
+    Notes
+    -----
+    The entropy of an ideal electron gas is defined as
+
+    .. math::
+        I_n(\mu,\beta) = \int \mathrm{d}\epsilon \epsilon^{n/2}
```
Comment (DanielKotik) [inline atoMEC/mathtools.py:265]: The integral limits are fix, right?
```suggestion
        I_n(\mu,\beta) = \int_0^\infty \mathrm{d}\epsilon\ \epsilon^{n/2}
```

File: atoMEC/mathtools.py [hunk 2]
```diff
@@ -194,6 +239,46 @@ def fd_int_complete(mu, beta, n):
...
+
+    where :math:`f_\mathrm{fd}=f_\mathrm{fd}(\epsilon,\mu,\beta)` is the Fermi-Dirac
+    distribution.
+    """
+    # use scipy quad integration routine
+    limup = np.inf
+
+    # ignore integration warnings (omnipresent because of inf upper limit)
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore")
```
Comment (DanielKotik) [inline atoMEC/mathtools.py:276]: I am not sure about this. Suppress warnings only when there is a good reason for that. For which parameter range `quad` actually raises these? If this is due to singularities within the integrand, then specify these with the `points` parameter. 

I also recommend adding tests in general to test expected outcome for a specific parameter set (including edge cases).

**Review by DanielKotik — CHANGES_REQUESTED** [review]

**Comment by DanielKotik** [comment]
> @DanielKotik I can confirm these changes are ok from the "physics perspective" but I'm not sure why one of the checks is failing.

Deploying the Sphinx documentation to GitHub pages fails due to a permission errors. I guess this is related to the fact that this PR comes from a forked repo - the PR triggers the pipline but deploying to GitHub pages is restricted and will only happen when the PR got merged.

File: atoMEC/mathtools.py (hunk 2 above)
Comment (timcallow) [reply atoMEC/mathtools.py:276]: Fine by me to do it this way. Just to note that there is also the same warning suppression in the `fd_int_complete` function. 

So my question is, should we deal only with the newly introduced warning in the current PR, or both? Or, merge this PR as is and then open a new one to address both issues simultaneously?

Screenshot from 2021-07-26 09-23-36

Comment (DanielKotik) [reply atoMEC/mathtools.py:276]: Well, then lets merge and open up a new issue on the warnings.

**Review by DanielKotik — APPROVED** [review]
